internal/provider/vaultsecrets/resource_vault_secrets_integration_postgres.go

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vaultsecrets
+
+import (
+	"context"
+	"fmt"
+	"maps"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/client/secret_service"
+	secretmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-vault-secrets/preview/2023-11-28/models"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-provider-hcp/internal/clients"
+	"github.com/hashicorp/terraform-provider-hcp/internal/provider/modifiers"
+)
+
+type IntegrationPostgres struct {
+	// Input fields
+	ProjectID               types.String `tfsdk:"project_id"`
+	Name                    types.String `tfsdk:"name"`
+	Capabilities            types.Set    `tfsdk:"capabilities"`
+	StaticCredentialDetails types.Object `tfsdk:"static_credential_details"`
+
+	// Computed fields
+	OrganizationID types.String `tfsdk:"organization_id"`
+	ResourceID     types.String `tfsdk:"resource_id"`
+	ResourceName   types.String `tfsdk:"resource_name"`
+
+	// Inner API-compatible models derived from the Terraform fields
+	capabilities            []*secretmodels.Secrets20231128Capability                     `tfsdk:"-"`
+	staticCredentialDetails *secretmodels.Secrets20231128PostgresStaticCredentialsRequest `tfsdk:"-"`
```

Comment:
We may need to add a gateway pool ID here as well

Reply:
should we introduce that separately (in a diff PR) or here?
are (should) gateways be added to the tf provider?

Reply:
I'd opt for here since a postgres integration is not particularly useful without a gateway